src/Symfony/Component/Translation/Translator.php

```diff
@@ -372,6 +372,39 @@ private function doLoadCatalogue($locale)
                 $this->catalogues[$locale]->addCatalogue($this->loaders[$resource[0]]->load($resource[1], $locale, $resource[2]));
             }
         }
+
+        $this->resolveCatalogueMessages($this->catalogues[$locale]);
+    }
+
+    private function resolveCatalogueMessages($catalogue)
+    {
+        $expr = '/{([^}]++)}/';
+        foreach ($catalogue->all() as $domain => $messages) {
+            preg_match_all($expr, implode($messages), $matches);
+            $parameters = [];
+            foreach ($matches[1] as $key => $id) {
+                if (isset($messages[$id])) {
+                    $parameters[$matches[0][$key]] = $messages[$id];
+                }
+            }
+
+            while (preg_match($expr, implode($parameters))) {
+                foreach ($parameters as $id => $message) {
+                    $parameters[$id] = str_replace(array_keys($parameters), array_values($parameters), $message);
```

Comment:
array_keys should be computed outside of the loop, and arra_values is not required isn't it?